x-pack/legacy/plugins/apm/public/components/shared/MetadataTable/helper.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { get, pick, isEmpty } from 'lodash';
+import { Section as SectionType } from './sections';
+import { Transaction } from '../../../../typings/es_schemas/ui/Transaction';
+import { APMError } from '../../../../typings/es_schemas/ui/APMError';
+import { Span } from '../../../../typings/es_schemas/ui/Span';
+import { flattenObject, KeyValuePair } from '../../../utils/flattenObject';
+
+export type MetadataItems = Array<SectionType & { rows?: KeyValuePair[] }>;
```

Comment:
I still find these "item" names pretty confusing since it's not clear what's a section and what's a row. So I think it's worth being explicit about this and calling it eg. `SectionsWithRows` (I wanted to call it `Sections` but we already have a definition for a `Section` that doesn't have rows).

Additionally the generated type is currently not super readable:
```ts
type MetadataItems = (SectionType & {
    rows?: KeyValuePair[] | undefined;
})[]
```

Instead I suggest to infer it:

```ts
type SectionsWithRows = ReturnType<typeof getMetadataItems>;
```

which will give you a more readable type:
```ts
type SectionsWithRows = {
    rows: KeyValuePair[];
    key: string;
    label: string;
    required?: boolean | undefined;
    properties?: string[] | undefined;
}[]
```

Or alternatively define it like:
```ts
interface SectionWithRows {
  key: string;
  label: string;
  rows: KeyValuePair[];
}
```

This way it's defined as singular, and the unneeded `required` and `properties` are removed.